frankenphp.c

```diff
@@ -105,13 +106,23 @@ static uintptr_t frankenphp_clean_server_context() {
 
 static void frankenphp_request_reset() {
   zend_try {
-    int i;
+    frankenphp_server_context *ctx = SG(server_context);
 
-    for (i = 0; i < NUM_TRACK_VARS; i++) {
-      zval_ptr_dtor(&PG(http_globals)[i]);
-    }
+    if (ctx->worker_ready) {
+      int i;
+      for (i = 0; i < NUM_TRACK_VARS; i++) {
+        zval_ptr_dtor(&PG(http_globals)[i]);
+      }
 
-    memset(&PG(http_globals), 0, sizeof(zval) * NUM_TRACK_VARS);
+      // Restore worker script super globals
+      memcpy(&PG(http_globals), ctx->worker_http_globals,
+             sizeof(zval) * NUM_TRACK_VARS);
+      php_hash_environment();
+    } else {
+      ctx->worker_http_globals = (zval **)malloc(sizeof(zval) * NUM_TRACK_VARS);
+      memcpy(ctx->worker_http_globals, &PG(http_globals),
+             sizeof(zval) * NUM_TRACK_VARS);
```

Comment:
And here it probably should be this (untested):
```suggestion
      memcpy(ctx->worker_http_globals, PG(http_globals),
             sizeof(zval) * NUM_TRACK_VARS);
```